### PR TITLE
Print an error message and exit if docker build failed

### LINKED ISSUE #99: Error message for failed docker build is a bit confusing
I made a mistake in the `Dockerfile` that I tried to deploy as a serverless container and the error message from the `serverless deploy` command was this:

```
Using credentials from serverless.yml
Updating container app...
Building and pushing container app to: rg.nl-ams.scw.cloud/xxxx/app:latest ...
Deploying Containers...
Waiting for container deployments, this may take multiple minutes...
Environment: linux, node 14.20.0, framework 3.21.0 (local), plugin 6.2.2, SDK 4.3.2
Docs:        docs.serverless.com
Support:     forum.serverless.com
Bugs:        github.com/serverless/serverless/issues
Error:
Error: Error: Error: Error: Unable to fetch image "rg.nl-ams.scw.cloud/xxxx/app:latest": failed to resolve image to digest: HEAD https://rg.nl-ams.scw.cloud/v2/xxxx/app/manifests/latest: unexpected status code 404 Not Found (HEAD responses have no body, use GET for details).
    at manageError (/builds/xxxx/xxxx/node_modules/serverless-scaleway-functions/shared/api/utils.js:20:11)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
```

This is a bit confusing, because the logging suggests that the build and push succeeded and the deployment after that was is the problem.
Only after I activated `--verbose` it became clear to me that the docker build was the problem.

### PR DESCRIPTION
**Pull request by norbjd**
Before this PR, if the docker build command failed, but a previous build has succeeded, serverless deploy would deploy the previously built image instead of just throwing an error.

This PR aims to solve this by printing an error message and exiting if the build failed for some reason.

Fixes #99

### CODE REVIEW

**Review by Shillaker — APPROVED** [review]

File: deploy/lib/buildAndPushContainers.js [hunk 1]
```diff
@@ -5,15 +5,37 @@ const tar = require('tar-fs');
...
   });
-  stream.on('end', resolve);
-  stream.on('error', reject);
-});
+}
+
+function findErrorInBuildOutput(buildOutput) {
+  for (let i = 0; i < buildOutput.length; i++) {
+    if (buildOutput[i].startsWith('{"errorDetail":{')) {
+      const errorDetail = JSON.parse(buildOutput[i])['errorDetail'];
```
Comment (thomas-tacquet) [inline deploy/lib/buildAndPushContainers.js:31]: JSON.parse [can throw exceptions](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse#exceptions)

can you surround it with try/catch ?

```javascript
let errorDetail;
try {
errorDetail = JSON.parse(buildOutput[i])['errorDetail'];
}
catch(err) {
return "" // (or something undefinied ?)
}
try {

```

File: deploy/lib/buildAndPushContainers.js [hunk 2]
```diff
@@ -32,12 +54,18 @@ module.exports = {
...
+        if (buildError !== undefined) {
+          reject("Build did not succeed, error: "+buildError);
+          return
+        }
 
         const image = docker.getImage(imageName)
 
         const inspectedImage = await image.inspect()
-          .catch(() => reject("Error during build of the image "+imageName+": run --verbose to see the error"));
+          .catch(() => reject("Image "+imageName+" does not exist: run --verbose to see errors"));
```
Comment (almottier) [inline deploy/lib/buildAndPushContainers.js:68]: you may use template literal with backticks when formatting string in js :
```js
`string text ${expression} string text`
```

File: deploy/lib/buildAndPushContainers.js [hunk 3]
```diff
@@ -5,15 +5,37 @@ const tar = require('tar-fs');
...
+    });
+    stream.on('error', reject);
   });
-  stream.on('end', resolve);
-  stream.on('error', reject);
-});
+}
+
+function findErrorInBuildOutput(buildOutput) {
+  for (let i = 0; i < buildOutput.length; i++) {
```
Comment (almottier) [inline deploy/lib/buildAndPushContainers.js:29]: why not use `forEach` ?

File: deploy/lib/buildAndPushContainers.js (hunk 1 above)
Comment (almottier) [inline deploy/lib/buildAndPushContainers.js:31]: you may need to catch exception

Comment (norbjd) [reply deploy/lib/buildAndPushContainers.js:31]: I've added a try/catch :)

File: deploy/lib/buildAndPushContainers.js (hunk 3 above)
Comment (norbjd) [reply deploy/lib/buildAndPushContainers.js:29]: We can't return inside a `forEach`, but I've changed to `for (const ... of ...)`.

File: deploy/lib/buildAndPushContainers.js (hunk 1 above)
Comment (norbjd) [reply deploy/lib/buildAndPushContainers.js:31]: https://github.com/scaleway/serverless-scaleway-functions/pull/102#discussion_r958191164 :smile:

**Review by thomas-tacquet — APPROVED** [review]

**Pull request merged**